### PR TITLE
HPCC-13545 Expose lock info to clients.

### PR DESCRIPTION
**Pull request by jakesmith**
Previously locking information was available only en-bulk via
a single output processed by Dali and returned as a formatted
string table.
There was a couple of places (in daliadmin and dalidiag) that
worked hard to parse this formatted string, to perform
filtering.

This change allows path and ip filtering of locks to be
handled at the server, for the result to be serialized to the
requesting client and then deserialized into interrogatable
interfaces.
This will also allow clients (e.g. Eclwatch) to request and
format the locking information as they see fit.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@wangkx - please review.

NB: Left as 2 commits, as 1st is just a search and replace rename, which will confuse reviewing of the 2nd commit, so I suggest reviewing separately.

**Comment by wangkx** [comment]
@jakesmith back to you.

**Comment by jakesmith** [comment]
@wangkx - finally got around to replying to your comments

**Comment by wangkx** [comment]
@jakesmith thanks for the reply. The code looks fine for me.
